core/server/master/src/main/java/alluxio/master/DefaultAlluxioMaster.java

```diff
@@ -262,33 +234,19 @@ public void stop() throws Exception {
   protected void startMasters(boolean isLeader) {
     try {
       connectToUFS();
-
-      mBlockMaster.start(isLeader);
-      mFileSystemMaster.start(isLeader);
-      if (LineageUtils.isLineageEnabled()) {
-        mLineageMaster.start(isLeader);
-      }
-      // start additional masters
-      for (Master master : mAdditionalMasters) {
+      for (Master master : mRegistry.getMasters()) {
```

Comment:
for example, I thought file master depends on metadata managed by block master, but after second thought, starting a file master without using the file master service would not query block master, so it's OK.